src/main/java/graphql/nadel/engine/Execution.java

```diff
@@ -87,13 +88,15 @@ public Execution(List<Service> services, GraphQLSchema overallSchema, NadelInstr
         InstrumentationContext<ExecutionResult> instrumentationCtx = instrumentation.beginExecute(new NadelInstrumentationExecuteOperationParameters(executionContext, instrumentationState));
 
         CompletableFuture<ExecutionResult> result;
+        ResultComplexityAggregator resultComplexityAggregator = new ResultComplexityAggregator(0, new LinkedHashMap<>());
         if (introspectionRunner.isIntrospectionQuery(executionContext, fieldSubSelection)) {
             result = introspectionRunner.runIntrospection(executionContext, fieldSubSelection, executionInput);
         } else {
-            CompletableFuture<RootExecutionResultNode> resultNodes = nadelExecutionStrategy.execute(executionContext, fieldSubSelection);
+            CompletableFuture<RootExecutionResultNode> resultNodes = nadelExecutionStrategy.execute(executionContext, fieldSubSelection, resultComplexityAggregator);
             result = resultNodes.thenApply(rootResultNode -> {
                 rootResultNode = instrumentation.instrumentRootExecutionResult(rootResultNode, new NadelInstrumentRootExecutionResultParameters(executionContext, instrumentationState));
-                return ResultNodesUtil.toExecutionResult(rootResultNode);
+                ExecutionResult executionResult = withNodeComplexity(ResultNodesUtil.toExecutionResult(rootResultNode), resultComplexityAggregator);
```

Comment:
@bbakerman any reason why we don't implement this as an instrumentation? It seems possible with `instrumentRootExecutionResult` and `beginFieldFetch` if we enable `FieldTracking`; all the hydrate calls go through `FieldTracking` from what I can see.

That way, it would be possible to move the instrumentation to the gateway rather than bloating Nadel; granted we merge in the changes to `ExecutionResultNode#getResultNodeCount` etc here.